var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -207,9 +241,11 @@ def setup_build_environment(self, env):
 
         if "+mpi" in spec:
             env.set("ESMF_CXX", spec["mpi"].mpicxx)
+            env.set("ESMF_C", spec["mpi"].mpicc)
```

Comment:
Maybe this helps with the problems on Narwhal ...

Reply:
I doubt it will. This is just to make sure the C compiler is set up correctly ... which is important for the ESMF develop branch where we make a cleaner distinction between code that's pure C vs C++. It should not affect the C++ compilation/linking that was the issue on your latest Narwhal Spack builds. But it also should not hurt.